app/controllers/users_controller.rb

```diff
@@ -1,3 +1,31 @@
 class UsersController < ApplicationController
-  def new; end
+  def show
+    @user = User.find_by(id: params[:id])
+    if @user.nil?
+      redirect_to root_url
+      flash[:danger] = t "not_found"
+    end
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash.now[:success] = t "welcome"
+      redirect_to @user
+    else
+      flash.now[:danger] = t "fail"
+      render :new
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:name, :email, :password,
```

Comment:
trong model khai báo như này
USER_ATTR = %i(name email password password_confirmation)
sau đó thay đổi như này
```suggestion
    params.require(:user).permit User::USER_ATTR
```